### PR TITLE
Beta-Binomial distribution, and related functions

### PR DESCRIPTION
**Pull request by ajwillshire**
Adding the Beta-Binomial distribution. [https://en.wikipedia.org/wiki/Generalized_hypergeometric_function](https://en.wikipedia.org/wiki/Beta-binomial_distribution)

Also added the Generalized Hypergeometric Function as I thought it was required for calculating the CDF. However, most implementations just traverse over the PMF and sum. (Plus, I don't think Wikipedia is correct in the stated formula for the CDF). [https://en.wikipedia.org/wiki/Generalized_hypergeometric_function](https://en.wikipedia.org/wiki/Generalized_hypergeometric_function)

This also required adding the Rising and Falling Factorials [https://en.wikipedia.org/wiki/Falling_and_rising_factorials](https://en.wikipedia.org/wiki/Falling_and_rising_factorials)

### CODE REVIEW

File: src/Numerics/Distributions/BetaBinomial.cs [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+
+        public System.Random RandomSource
+        {
+            get => _random;
+            set => _random = value ?? SystemRandomSource.Default;
+        }
+        /// <summary>
+        /// Gets the mean of the distribution.
+        /// </summary>
+        double IUnivariateDistribution.Mean => (_n * _a) / (_a + _b);
```
Comment (cdrnet) [inline src/Numerics/Distributions/BetaBinomial.cs:144]: What is the reasoning for choosing explicit interface implementations here, i.e. for making them unavailable on the type?

**Comment by cdrnet** [comment]
Looks great, thanks!

**Comment by ajwillshire** [comment]
No probs.

Re the question of explicit interface implementations, I just tried to copy the existing style of (I think) the Negative Binomial distribution. I'm a bit out of my comfort zone, so happy for changes to be made as required.

Thanks!

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by cdrnet** [comment]
No worries, I can go over them - the explicit interface implementation actually makes sense for those throwing exceptions. Unfortunately we cannot do that on the other existing ones as this would be a breaking change.

**Pull request merged**